File: sic-api/Entities/Db/DbTitle.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;
using Microsoft.EntityFrameworkCore;
using sic_api.Entities;

namespace sic_api.Entities.Db;

[Table("db_title")]
public class DbTitle : BaseEntity
{

    [Required]
    [MaxLength(20)]
    [Column("person_type")]
    public string PersonType { get; set; } = default!;

    [Required]
    [MaxLength(100)]
    [Column("prefix_short_name_en")]
    public string PrefixShortNameEn { get; set; } = default!;


    [Required]
    [MaxLength(100)]
    [Column("prefix_short_name_local")]
    public string PrefixShortNameLocal { get; set; } = default!;


    [MaxLength(100)]
    [Column("suffix_short_name_en")]
    public string? SuffixShortNameEn { get; set; }


    [MaxLength(100)]
    [Column("suffix_short_name_local")]
    public string? SuffixShortNameLocal { get; set; }

    [Required]
    [MaxLength(100)]
    [Column("prefix_name_en")]
    public string PrefixNameEn { get; set; } = default!;


    [Required]
    [MaxLength(100)]
    [Column("prefix_name_local")]
    public string PrefixNameLocal { get; set; } = default!;


    [MaxLength(100)]
    [Column("suffix_name_en")]
    public string? SuffixNameEn { get; set; }


    [MaxLength(100)]
    [Column("suffix_name_local")]
    public string? SuffixNameLocal { get; set; }

    [Column("sort_order")]
    public int? SortOrder { get; set; }

    [Required]
    [Column("is_active")]
    public bool IsActive { get; set; } = false;
}

File: sic-spring/sic/src/main/java/com/softinter/sicapi/entity/db/DbTitle.java
package com.softinter.sicapi.entity.db;

import com.softinter.sicapi.entity.base.BaseEntity;
import jakarta.persistence.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "db_title")
@Data
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
public class DbTitle extends BaseEntity {

    @Column(name = "person_type", nullable = false, length = 20)
    private String personType;

    @Column(name = "prefix_short_name_en", nullable = false, length = 100)
    private String prefixShortNameEn;

    @Column(name = "prefix_short_name_local", nullable = false, length = 100)
    private String prefixShortNameLocal;

    @Column(name = "suffix_short_name_en", length = 100)
    private String suffixShortNameEn;

    @Column(name = "suffix_short_name_local", length = 100)
    private String suffixShortNameLocal;

    @Column(name = "prefix_name_en", nullable = false, length = 100)
    private String prefixNameEn;

    @Column(name = "prefix_name_local", nullable = false, length = 100)
    private String prefixNameLocal;

    @Column(name = "suffix_name_en", length = 100)
    private String suffixNameEn;

    @Column(name = "suffix_name_local", length = 100)
    private String suffixNameLocal;

    @Column(name = "sort_order")
    private Integer sortOrder;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = false;
}